### PR TITLE
API: Make BasicAuth Prefix case insensitive

### PR DESCRIPTION
**Pull request by 6543**
Before:
 * `curl -X POST "http://gitea.com/api/v1/... -H  "accept: application/json" -H  "authorization: Basic dTE6MTIzMTIz" ...`
 * `curl -X POST "http://gitea.com/api/v1/... -H  "accept: application/json" -H  "Authorization: Basic dTE6MTIzMTIz" ...`

After:
 * `curl -X POST "http://gitea.com/api/v1/... -H  "accept: application/json" -H  "authorization: Basic dTE6MTIzMTIz" ...`
 * `curl -X POST "http://gitea.com/api/v1/... -H  "accept: application/json" -H  "authorization: basic dTE6MTIzMTIz" ...`
 * `curl -X POST "http://gitea.com/api/v1/... -H  "accept: application/json" -H  "Authorization: basic dTE6MTIzMTIz" ...`
 * `curl -X POST "http://gitea.com/api/v1/... -H  "accept: application/json" -H  "Authorization: Basic dTE6MTIzMTIz" ...`

### CODE REVIEW

File: modules/auth/sso/basic.go [hunk 1]
```diff
@@ -52,11 +52,14 @@ func (b *Basic) IsEnabled() bool {
...
-		return nil
+		baHead = ctx.Req.Header.Get("authorization")
+		if len(baHead) == 0 {
+			return nil
+		}
 	}
 
 	auths := strings.Fields(baHead)
-	if len(auths) != 2 || auths[0] != "Basic" {
+	if len(auths) != 2 || strings.ToLower(auths[0]) != "basic" {
```
Comment (lunny) [inline modules/auth/sso/basic.go]: `(auths[0] != "Basic" && auths[0] != "basic")` Otherwise it will allow `bAsic`.

Comment (6543) [inline modules/auth/sso/basic.go]:
```suggestion
	if len(auths) != 2 || (auths[0] != "Basic" && auths[0] != "basic") {
```

**Review by lunny — APPROVED** [review]

File: modules/auth/sso/basic.go [hunk 2]
```diff
@@ -52,11 +52,14 @@ func (b *Basic) IsEnabled() bool {
 func (b *Basic) VerifyAuthData(ctx *macaron.Context, sess session.Store) *models.User {
 	baHead := ctx.Req.Header.Get("Authorization")
 	if len(baHead) == 0 {
-		return nil
+		baHead = ctx.Req.Header.Get("authorization")
```
Comment (lafriks) [inline modules/auth/sso/basic.go]: This is not needed. Header.Get is case insensitivie

**Review by lafriks — DISMISSED** [review]
See https://play.golang.org/p/crtMhYsMDGY

File: modules/auth/sso/basic.go (hunk 2 above)
Comment (lafriks) [reply modules/auth/sso/basic.go]: It does already work with lowercase header name:
image

**Comment by 6543** [comment]
@lafriks done

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**